src/components/DebouncedTextField.tsx

```diff
@@ -33,6 +33,7 @@ interface Props {
   rows?: number;
   dispatchCallback?: (value: ValueType) => void;
   initialValue?: ValueType;
+  reset?: boolean;
```

Comment:
This smells of incorrect split of responsibilities. If we need to control internal state of the component from the outside it probably means that the state is not really that local.

I'd rather suggest to have `transactionId` being either passed as a prop or obtained from a reducer. Every time we send transaction `transactionId` is incremented, and all components are able to do any clean up they want if they notice that `transactionId` changes. On the first sight this looks much cleaner to me than the `reset/setToReset/toRest/previousReset` mumbojumbo happening internally now and is probably more future proof than passing `reset` to some specific components.

Reply:
shit, this is  such a great suggestion .... asking the transactionId to be different is such a greater event than transaction changed from false to true.